Mathlib/Tactic/ApplyAt.lean

```diff
@@ -0,0 +1,61 @@
+/-
+Copyright (c) 2023 Adam Topaz. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Adam Topaz
+-/
+import Lean
+
+/-!
+# Apply at
+
+A tactic for applying functions at hypotheses.
+-/
+
+open Lean Meta Elab Tactic Term
+
+/--
+This function is similar to `forallMetaTelescopeReducingUntilDefEq` except that
+it will construct mvars until it reaches one whose type is defeq to the given
+type `t`. It uses `forallMetaTelescopeReducing`.
+-/
+def Lean.Meta.forallMetaTelescopeReducingUntilDefEq
+    (e t : Expr) (kind : MetavarKind := MetavarKind.natural) :
+    MetaM (Array Expr × Array BinderInfo × Expr) := do
```

Comment:
This function has nothing to do here, it should be in some utility file, ultimately scheduled for move to sdt4.

Reply:
I agree. Do you have a suggestion for where it should be moved? Perhaps in `Mathlib/Lean/Meta/Basic`, given that `forallMetaTelescopeReducing` appears in `Lean/Meta/Basic` in core?

Reply:
Ok, moved to `Mathlib/Lean/Meta/Basic`.